core/src/lib/lex.cc

```diff
@@ -549,6 +552,53 @@ static uint64_t scan_pint64(LEX* lf, char* str)
   return val;
 }
 
+class M {
+ public:
+  M(FILE* fd) : pm(GetPoolMemory(PM_NAME)), fd_(fd) { pos_ = ftell(fd_); }
+  ~M()
+  {
+    FreePoolMemory(pm);
+    fseek(fd_, pos_, SEEK_SET);
+  }
+  POOLMEM* pm;
+
+ private:
+  FILE* fd_;
+  long pos_;
+};
+
+static bool NextLineContinuesWithQuotes(LEX* lf)
+{
+  M m(lf->fd);
+
+  if (bfgets(m.pm, lf->fd) != NULL) {
+    int i = 0;
+    while (m.pm[i]) {
+      if (m.pm[i] == '"') {
+        return true;
+      } else if (m.pm[i] != ' ') {
+        return false;
+      }
+      ++i;
+    };
```

Comment:
I'd compare to '\0' to clarify that this is an end-of-string check.
```suggestion
    for (int i = 0; m.pm[i] != '\0'; ++i) {
      if (m.pm[i] == ' ') { continue; } /* skip spaces */
      return (m.pm[i] == '"'); /* is the first non-space a double quote? */
    }```